src/Adapter/Product/QueryHandler/GetRelatedProductsHandler.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\QueryHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Query\GetRelatedProducts;
+use PrestaShop\PrestaShop\Core\Domain\Product\QueryHandler\GetRelatedProductsHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\QueryResult\RelatedProduct;
+
+final class GetRelatedProductsHandler implements GetRelatedProductsHandlerInterface
+{
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @param ProductRepository $productRepository
+     * @param int $contextLangId
+     */
+    public function __construct(
+        ProductRepository $productRepository,
+        int $contextLangId
+    ) {
+        $this->productRepository = $productRepository;
+        $this->contextLangId = $contextLangId;
```

Comment:
I wonder if we should inject this language in the handler Or handle it via the command instead

It would be more flexible then I guess, besides when we create the API that uses this command, the controller can easily fetch the context language, so it won't force us to post the language ID via ajax

However it allows to easily adapt this command to more use cases, whereas changing the context language is no easy task